src/common/formatFamiliesData.ts

```diff
@@ -1,28 +1,77 @@
 import { Person } from "@/components/Form/formFunctions";
 import { Schema } from "@/databaseUtils";
 import { displayList } from "@/common/format";
+import { getChildAgeUsingBirthYearAndMonth, isChildUsingBirthYear } from "@/common/getAgesOfFamily";
+import { getCurrentYear } from "@/common/date";
 
 export interface HouseholdSummary {
     householdSize: string;
     genderBreakdown: string;
+    ageAndGenderOfAdults: string;
     numberOfBabies: string;
     ageAndGenderOfChildren: string;
 }
 
 const getChild = (child: Person): string => {
-    const gender = child.gender === "male" ? "M" : child.gender === "female" ? "F" : "-";
-    return `${child.age} ${gender}`;
+    let gender;
+    switch (child.gender) {
+        case "male":
+            gender = "M";
+            break;
+        case "female":
+            gender = "F";
+            break;
+        case "other":
+            gender = "-";
+            break;
+    }
+
+    const age = getChildAgeUsingBirthYearAndMonth(child.birthYear, child.birthMonth ?? null, true);
+
+    return `${age} ${gender}`;
+};
+
+const getAdult = (adult: Person): string => {
+    let gender;
+    switch (adult.gender) {
+        case "male":
+            gender = "M";
+            break;
+        case "female":
+            gender = "F";
+            break;
+        case "other":
+            gender = "-";
+            break;
+    }
+    const age = getCurrentYear() - adult.birthYear;
+    return `${age} ${gender}`;
 };
 
 const convertPlural = (value: number, description: string): string => {
     return `${value} ${description}${value !== 1 ? "s" : ""}`;
 };
 
 export const prepareHouseholdSummary = (familyData: Schema["families"][]): HouseholdSummary => {
-    const children = familyData.filter((member) => member.age !== null);
-
+    const children = familyData.filter((member) => isChildUsingBirthYear(member.birth_year));
+    const formattedChildren: Person[] = children.map((child) => {
+        return {
+            gender: child.gender,
+            birthMonth: child.birth_month,
+            birthYear: child.birth_year,
+        };
+    });
+    const adults = familyData.filter((member) => !isChildUsingBirthYear(member.birth_year));
+    const formattedAdults: Person[] = adults.map((adult) => {
+        return {
+            gender: adult.gender,
+            birthYear: adult.birth_year,
+        };
+    });
```

Comment:
Wonder if you can pull this out into a function and pass in the function as a parameter for the filter (i.e. isChildUsingBirthYear - if passing in !isChildUsingBirthYear doesn't work you could make `isAdultUsingBirthYear(year) => !isChildUsingBirthYear(year)`)

Reply:
I have created two new functions isChildPerson, isAdultPerson because isChildUsingBirthYear takes in birth_year but we need to take in the person object in the filter.

Reply:
Oh I meant take the entire thing so something like 
```
const getFormattedPeople= (familyData: Schema["families"], filterFunction: (arg: Array) => Array): Person[] => {
    const people = familyData.filter((member) => filterFunction(member));
    return children.map((child) => {
        return {
            gender: child.gender,
            birthMonth: child.birth_month,
            birthYear: child.birth_year,
        };
    });
};
```
then call it using 
```
const formattedChildren = getFormattedPeople(familyData, isChildPerson);
const formattedAdults =  getFormattedPeople(familyData, isAdultPerson);
```
assuming this works? And then `birthMonth` is null/undefined in either case right? Could add a param to determine that

Reply:
I guess birthMonth will depend on if they have a birthMonth stored. If they do, it will be used to calculate the childs age (eg. if they are under 1 years old).